aya/src/maps/lpm_trie.rs

```diff
@@ -0,0 +1,456 @@
+//! A LPM Trie.
+use std::{convert::TryFrom, marker::PhantomData, mem, ops::Deref};
+
+use crate::{
+    generated::bpf_map_type::BPF_MAP_TYPE_LPM_TRIE,
+    maps::{IterableMap, Map, MapError, MapRef, MapRefMut},
+    sys::{bpf_map_delete_elem, bpf_map_lookup_elem, bpf_map_update_elem},
+    Pod,
+};
+
+/// A Longest Prefix Match Trie.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 4.20.
+///
+/// # Examples
+/// ```no_run
+/// # let bpf = aya::Bpf::load(&[])?;
+/// use aya::maps::lpm_trie::{LpmTrie, Key};
+/// use std::convert::TryFrom;
+/// use std::net::Ipv4Addr;
+///
+/// let mut trie = LpmTrie::try_from(bpf.map_mut("LPM_TRIE")?)?;
+/// let ipaddr = Ipv4Addr::new(8,8,8,8);
+/// // The following represents a key for the "8.8.8.8/16" subnet.
+/// // The 'prefix_len' field represents how many bytes should be matched against
+/// // and the data represents the actoual data to be matched.
+/// let key = Key::new(16, u32::from(ipaddr).to_be());
+/// trie.insert(&key, 1, 0)?;
+///
+/// // LPMTrie matches against the longest (most accurate) key.
+/// let lookup = Key::new(32, u32::from(ipaddr).to_be());
```

Comment:
It's not clear from the example, but for lookups is the key always 32 (/32) for IPv4?
Would I get any matches if I used `/8` for example? (I think the answer is no)

Depending on what you I think this may needs to say something like
`// LPMTrie::get() finds the longest matching prefix up to prefix_len bytes`

Reply:
For lookup you would need at least `prefix_len` bytes ...